src/sys/stm_tests.ml

```diff
@@ -0,0 +1,242 @@
+          then fs
+          else
+            let new_map = Map_names.remove next_in_path d.fs_map in
+            let new_map = Map_names.add next_in_path nfs new_map in
+            Directory {d with fs_map = new_map}))
+
+  let readdir_model fs path =
+    match find_opt_model fs path with
+    | None    -> None
+    | Some fs ->
+      match fs with
+      | File _ -> None
+      | Directory d -> Some (Map_names.fold (fun k _ l -> k::l) d.fs_map [])
+
+  let rec rmdir_model fs path delete_dir_name =
+    match fs with
+    | File _      -> fs
+    | Directory d ->
+      (match path with
+      | [] ->
+        (match Map_names.find_opt delete_dir_name d.fs_map with
+        | Some (Directory target) when Map_names.is_empty target.fs_map ->
+          Directory {d with fs_map = Map_names.remove delete_dir_name d.fs_map}
+        | None | Some (File _) | Some (Directory _) -> fs)
+      | next_in_path :: tl_path ->
+        (match Map_names.find_opt next_in_path d.fs_map with
+        | None        -> fs
+        | Some sub_fs ->
+          let nfs = rmdir_model sub_fs tl_path delete_dir_name in
+          if nfs = sub_fs
+          then fs
+          else
+            Directory {d with fs_map = (update_map_name d.fs_map next_in_path nfs)}))
+
+  let rec touch_model fs path new_file_name perm =
+    match fs with
+    | File _      -> fs
+    | Directory d ->
+      (match path with
+      | [] ->
+        let new_file = File {perm} in
+        Directory {d with fs_map = Map_names.add new_file_name new_file d.fs_map}
+      | next_in_path :: tl_path ->
+        (match Map_names.find_opt next_in_path d.fs_map with
+        | None        -> fs
+        | Some sub_fs ->
+          let nfs = touch_model sub_fs tl_path new_file_name perm in
+          if nfs = sub_fs
+          then fs
+          else
+            Directory {d with fs_map = update_map_name d.fs_map next_in_path nfs}))
+
+  let next_state c fs =
+    match c with
+    | File_exists _path -> fs
+    | Mkdir (path, new_dir_name, perm) ->
+      if mem_model fs (path @ [new_dir_name])
+      then fs
+      else mkdir_model fs path new_dir_name perm
+    | Rmdir (path,delete_dir_name) ->
+      if mem_model fs (path @ [delete_dir_name])
+      then rmdir_model fs path delete_dir_name
+      else fs
+    | Readdir _path -> fs
+    | Touch (path, new_file_name, perm) ->
+      if mem_model fs (path @ [new_file_name])
+      then fs
+      else touch_model fs path new_file_name perm
+
+  let init_sut () =
+    match Sys.os_type with
+    (* | "Unix" -> ignore (Sys.command ("rm -rf " ^ (static_path / "sandbox_root") ^ " && mkdir " ^ (static_path / "sandbox_root"))) *)
+    | "Unix" -> ignore (Sys.command ("mkdir " ^ (static_path / "sandbox_root")))
+    (* | "Win32" ->
+      ignore (Sys.command (
+      "powershell -Command \"Remove-Item -Path " ^ (static_path / "sandbox_root") ^ " -Recurse -Force -ErrorAction Ignore \"
+       & mkdir " ^ (static_path / "sandbox_root"))) *)
```

Comment:
You can safely remove these commented out lines now

Reply:
Do the CI clean everything after a run? Imagine if a test crashes (with an assert false, for example), the filesys will not be removed. I can remove it locally, but does the CI automatically reset things?